modules/ximgproc/src/deriche_filter.cpp

```diff
@@ -0,0 +1,593 @@
+#include "precomp.hpp"
+#include "opencv2/highgui.hpp"
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+namespace cv {
+namespace ximgproc {
+
+/*
+If you use this code please cite this @cite deriche1987using
+Using Canny's Criteria to Derive a Recursively Implemented Optimal Edge Detector International Journal of Computer Vision,167-187 (1987)
+*/
+
+    class ParallelGradientDericheYCols : public ParallelLoopBody
+    {
+    private:
+        Mat &img;
+        Mat &im1;
```

Comment:
About You should use Mat img; without &.  I connot do it because i don't know how to use img.ptr<uchar>(0)